### PR TITLE
[TRTLLM-14765][feat] Clean up K3 FP8 weight-read env flags, make them default off prior to mergeback to main

### PR DESCRIPTION
**Pull request by moraxu**
@coderabbitai summary

**Note: The second commit https://github.com/NVIDIA/TensorRT-LLM/pull/17080/commits/ee8fead750dda3144bebb587349b6902dcfcde9b is only pre-commit application**

## Description

<p style="margin: 10px 0px 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">On<span> </span><tt>feat/kimi_k3</tt>, the K3 weight loader builds FP8 (e4m3, 128x128 block-scale) copies of the replicated BF16 weights to cut the per-decode-step HBM read (~109 GB/rank under attention DP, roughly halved). This is a perf optimization that is<span> </span><b>lossy relative to the BF16 checkpoint</b>, and it is currently<span> </span><b>default ON</b><span> </span>(SM100-gated), controlled by five interacting env vars in<span> </span><tt>tensorrt_llm/_torch/models/modeling_kimi_linear.py</tt><span> </span>(definitions ~L140-182, decision logic ~L2475-2560):</p><div class="table-wrap" style="margin: 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

Flag | Scope | Current default
-- | -- | --
KIMI_K3_FP8_WEIGHT_READ | Master switch; shared-expert + latent MLP projections | on (SM100 only)
KIMI_K3_FP8_WEIGHT_READ_KDA | KDA q/k/v/g/o projections (~61 GB of the ~109 GB/rank read; most precision-sensitive slice) | on
KIMI_K3_FP8_WEIGHT_READ_MLA | MLA q_a/q_b/o + output-gate projections (kv_a/kv_b stay BF16) | on
KIMI_K3_KDA_GLUE_FP8 | Fused KDA decode fast path on FP8 modules (prototype; +8–16% tps/user) | on
KIMI_K3_FP8_WEIGHT_READ_GATE_UP | Fused shared-expert gate_up_proj | on under attention DP, off under TP

</div><p style="margin: 10px 0px 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Sub-flags only act when the master is on;<span> </span><tt>KDA=0</tt><span> </span>also restores the BF16 wrapper decode fast path;<span> </span><tt>GLUE_FP8</tt><span> </span>requires the KDA FP8 read.</p><p style="margin: 10px 0px 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Anyone loading K3 on Blackwell today silently gets lossy FP8 copies of weights the checkpoint ships as BF16. All accuracy results measured with default settings so far have been measuring the FP8-read path (relevant to the suspected numeric issue behind tool-call flakiness; see<span> </span><a href="https://jirasw.nvidia.com/browse/TRTLLM-14764" title="Strict K3 accuracy validation with Kimi Vendor Verifier (KVV) suite" class="issue-link" data-issue-key="TRTLLM-14764" style="color: rgb(0, 82, 204); text-decoration: rgb(0, 82, 204); cursor: pointer;">TRTLLM-14764</a>).</p><h3 style="margin: 30px 0px 0px; padding: 0px; font-size: 16px; font-weight: 500; line-height: 1.25; letter-spacing: -0.006em; text-transform: none; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; orphans: 2; text-align: left; text-indent: 0px; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><a name="Scope" title="Follow link" style="color: rgb(0, 82, 204); text-decoration: rgb(0, 82, 204); cursor: pointer;"></a>Scope</h3><ol style="margin: 10px 0px 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li>Flip defaults to off: BF16 unless explicitly requested (<tt>"1"</tt><span> </span>-&gt;<span> </span><tt>"0"</tt><span> </span>defaults, drop the DP-dependent gate_up default).</li><li>Consolidate the flag sprawl before mainline merge: prefer a single opt-in knob with the per-slice flags as expert overrides at most.</li><li>Update touchpoints that assume default-on semantics:<span> </span><tt>examples/kimi_k3/perf_sweep/perf_sweep.sbatch</tt><span> </span>and<span> </span><tt>acc_sweep.sbatch</tt><span> </span>set flag combos explicitly.</li><li>TODO: Coordinate with perf owners: flipping the default silently costs ~10-15% decode tps for any recipe not setting the flags, so published perf configs need explicit opt-in or their numbers regress.</li></ol><p style="margin: 10px 0px 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">All flag usage is contained in three files (<tt>modeling_kimi_linear.py</tt><span> </span>+ the two sweep scripts)

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [ ] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Review by brnguyen2 — APPROVED** [review]

**Comment by brnguyen2** [comment]
Superseded by #17129 — closing this one in its favour.

Context, so nothing here looks like a judgement on the change itself: the only thing blocking this PR was the DCO check. The sign-off trailer on `83fd8032` has a machine hostname in place of the email address, so DCO stayed at `ACTION_REQUIRED`. (The author field on both commits was already correct; it was only the trailer.) Rather than ask for a history rewrite and force-push, I re-submitted the same work.

What #17129 is:

- Byte-identical file content to `ee8fead7` for all three files. I confirmed there was no drift in them between this PR's base and the current `feat/kimi_k3` tip, so they applied cleanly.
- Squashed into one commit with **@moraxu preserved as the author**, carrying your own `Signed-off-by: Michal Guzek <mguzek@nvidia.com>` (the trailer you wrote on `ee8fead7`), plus mine as the submitter. DCO passes.
- Rebased onto the current tip.
- Adds local validation to the description.

On the validation: rather than just check that the new default runs, I tested the claim that this is a pure default flip. Comparing the pre- and post-change trees on a truncated Kimi K3 checkpoint (TP4, attention DP, 36 prompts, full output tokens plus logits): toggling the flag on the pre-change tree changes 36/36 outputs, which confirms the FP8 conversion genuinely engages in that configuration; the post-change tree with `KIMI_K3_FP8_WEIGHT_READ=1` is identical to the pre-change default, and the post-change default is identical to the pre-change `KIMI_K3_FP8_WEIGHT_READ=0`. Both 36/36 identical, zero drift. So neither path's numerics moved and only the default selection changed, which is exactly what the change intends. Unit suites pass (76 passed, 1 skipped).

One unrelated thing worth knowing, since it could have sent you chasing a ghost: the CI failure analysis on the last run here reported "PR likely to blame? Yes" for `unittest/api_stability`. That looks wrong. `api_stability` imports the LLM API classes and the base checkpoint loader — it never imports `modeling_kimi_linear` — and the same failure occurs on other PRs to this branch that touch only docs and `requirements.txt`, where the same analysis tool reported "No". It appears to be a pre-existing failure on the branch rather than anything to do with this change.

Thanks — the change itself is unmodified, just re-packaged so it can merge.
